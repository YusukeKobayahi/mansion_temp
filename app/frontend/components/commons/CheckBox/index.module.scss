@import "~/styles/variables";
@import "~/styles/utils";

.checkbox {
  input[type="checkbox"] {
    display: none;
  }
  label {
    cursor: pointer;
    display: flex;
    text-align: center;
    align-items: center;
    justify-content: center;

    width: 26px;
    height: 26px;
    color: #fff;
    border: 2px solid;
    border-color: $color-main;
  }
}

.checked {
  background-color: $color-main;
}
