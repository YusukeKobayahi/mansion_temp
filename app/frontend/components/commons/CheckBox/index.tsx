import { ChangeEventHandler } from "react";

import styles from "~/components/commons/CheckBox/index.module.scss";

type CheckBoxProps = {
  disabled?: boolean;
  checked: boolean;
  value: string;
  onChange?: ChangeEventHandler<HTMLInputElement>;
};

const CheckBox: React.FC<CheckBoxProps> = ({
  checked,
  disabled = false,
  value,
  onChange,
}: CheckBoxProps) => {
  const checkboxId = `checkbox-${value}`;
  return (
    <div className={styles.checkbox}>
      <input
        id={checkboxId}
        type="checkbox"
        disabled={disabled}
        value={value}
        checked={checked}
        onChange={onChange}
      />
      <label htmlFor={checkboxId} className={checked ? styles.checked : ""}>
        {"✔︎"}
      </label>
    </div>
  );
};

export default CheckBox;
