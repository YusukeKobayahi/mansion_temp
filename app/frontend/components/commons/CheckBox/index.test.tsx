import { render } from "@testing-library/react";
import CheckBox from "~/components/commons/CheckBox";

describe("components/commons/CheckBox", () => {
  it("正常に表示される", () => {
    const { asFragment } = render(<CheckBox checked={false} value={"value"} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
