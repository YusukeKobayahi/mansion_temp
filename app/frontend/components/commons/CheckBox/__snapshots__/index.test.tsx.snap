// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/CheckBox 正常に表示される 1`] = `
<DocumentFragment>
  <div
    class="checkbox"
  >
    <input
      id="checkbox-value"
      type="checkbox"
      value="value"
    />
    <label
      class=""
      for="checkbox-value"
    >
      ✔︎
    </label>
  </div>
</DocumentFragment>
`;
